catalog/app/components/Preview/loaders/summarize.ts

```diff
@@ -0,0 +1,40 @@
+export type TypeShorthand =
+  | 'echarts'
+  | 'json'
+  | 'jupyter'
+  | 'perspective'
+  | 'vega'
+  | 'voila'
+
+export type FileShortcut = string
+
+export interface StyleOptions {
+  height?: string
+}
+
+export interface PerspectiveOptions {
+  settings?: boolean
+}
+
+export interface TypeExtended extends PerspectiveOptions {
```

Comment:
My plan was to add more specific types like this:

```ts
export interface TypeExtended extends PerspectiveOptions, EchartsOptions, VoilaOptions {}
```
You point, as I understand: it's not right because we can't use perspective and echarts in one object, and we should split perspective and echarts like this:

```ts
export type TypeExtended =
  | (TypeExtendedEssentials & PerspectiveOptions)
  | (TypeExtendedEssentials & EchartsOptions)
```

Reply:
yep, tho it probably should be more like `export type TypeExtended = TypeExtendedEssentials & (PerspectiveOptions | EchartsOptions)`, but you've got the idea right. your initial approach would allow representation of incorrect states, e.g. name=perspective + options from some other format, but using a discriminated union would prevent that